src/rpdk/core/contract/resource_client.py

```diff
@@ -186,6 +187,21 @@ def assert_write_only_property_does_not_exist(self, resource_model):
             ), "The model MUST NOT return properties defined as \
                 writeOnlyProperties in the resource schema"
 
+    def get_metadata(self, metadata_key):
+        metadata = []
+        try:
+            properties = self._schema["properties"]
+        except KeyError:
+            return metadata
+
+        for prop in properties:
+            try:
+                if properties[prop][metadata_key] == "false":
```

Comment:
we are returning properties that have the metadata_key and a specific value. Maybe we could add a parameter that takes the value as well.

Reply:
I am not sure I understand. Could you give me an example?

Reply:
```python
get_metadata(self, metadata_key, value):
     .
     .
     .
      if properties[prop][metadata_key] == value:

Reply:
maybe something like:
```python
metadata = set()

metadata.add(
[
    prop for prop in properties.keys() if "insertionOrder" in properties[prop] and properties[prop]["insertionOrder"] == "false"
])
```

Reply:
> ```python
> get_metadata(self, metadata_key, value):
>      .
>      .
>      .
>       if properties[prop][metadata_key] == value:
> ```

This is a valid point. This is useful in cases we have multiple metadata to check. In this case, I think we only care about insertion order. Might be worth removing the metadat_key too

Reply:
> maybe something like:
> 
> ```python
> metadata = set()
> 
> metadata.add(
> [
>     prop for prop in properties.keys() if "insertionOrder" in properties[prop] and properties[prop]["insertionOrder"] == "false"
> ])
> ```

will make the change